src/amo/reducers/versions.js

```diff
@@ -472,14 +472,25 @@ const reducer = (
         }
       }
 
-      for (const collection of collections) {
-        if (collection && collection.results) {
-          for (const addon of collection.results) {
-            const version = createInternalVersion(
-              addon.addon.current_version,
-              state.lang,
-            );
-            newVersions[version.id] = version;
+      for (const shelf of Object.keys(shelves)) {
+        if (shelves[shelf]) {
```

Comment:
This code in here looks like it's duplicated between here and the code that processes the `homeShelves`. It should be extracted in a way that it can be reused in both places.

Reply:
Nice work extracting this code!